rapid7_insightvm/komand_rapid7_insightvm/util/resource_helpers.py

```diff
@@ -0,0 +1,182 @@
+        """
+        endpoint = endpoints.Role.roles(console_url, role_id)
+        try:
+            response = self.requests.resource_request(endpoint=endpoint)
+            return response
+        except ResourceNotFound:
+            # Get roles to provide a better error
+            endpoint = endpoints.Role.roles(console_url)
+            roles = self.requests.resource_request(endpoint=endpoint)['resources']
+            role_ids = [r['id'] for r in roles]
+
+            error = f"ERROR: Specified role ID ({role_id}) does not exist, valid role IDs: {role_ids}"
+            self.logger.error(error)
+            raise ResourceNotFound(error)
+
+    def validate_user_permissions(self, console_url: str, user: dict) -> dict:
+        """
+        Validate that user permissions are appropriate given the user's role.
+        This is useful when updating a user's role or create a new user as
+        some roles required the allSites, allAssetGroups, or superuser parameters
+        to be set.
+        :param console_url: URL to the InsightVM console
+        :param user: dict containing the details for a user (from the InsightVM API or matching the format)
+        :return: The user dict with updated permissions if required
+        """
+        endpoint = endpoints.Role.roles(console_url, user['role']['id'])
+        role = self.requests.resource_request(endpoint=endpoint)
+
+        # Set All Sites permission if required and not set
+        if not user['role']['allSites']:
+            for permission in ("all-permissions", "manage-sites", "manage-tags"):
+                if permission in role['privileges']:
+                    self.logger.info(
+                        f"Setting 'Access All Sites' to 'true' as it is required based "
+                        f"on the permissions for the role '{user['role']['id']}'"
+                    )
+                    user['role']['allSites'] = True
+                    break
+
+        # Set All Asset Groups permission if required and not set
+        if not user['role']['allAssetGroups']:
+            for permission in ("all-permissions", "manage-dynamic-asset-groups"):
+                if permission in role['privileges']:
+                    self.logger.info(
+                        f"Setting 'Access All Asset Groups' to 'true' as it is required based "
+                        f"on the permissions for the role '{user['role']['id']}'"
+                    )
+                    user['role']['allAssetGroups'] = True
+                    break
+
+        # Set superuser permission if required and not set
+        # Do this silently as it isn't exposed to the user of the plugin and
+        # appears to simply be an internal setting that custom roles cannot have
+
+        if not user['role']['superuser'] and (user['role']['id'] == 'global-admin'):
+            user['role']['superuser'] = True
+
+        return user
+
+    def validate_user_email(self, email: str) -> None:
```

Comment:
Do you really want this function to raise an Exception? Or do you want the calling function to catch a boolean returned by this function and raise an exception based on the result? It feels like the latter is more usable.

Reply:
this function is only called by the validate_user function which is right below it so it dose not really make a difference